ref_arch/multi_tenant.rst

```diff
@@ -0,0 +1,540 @@
+
+::
+
+  -- transactionally remove campaign 46 and all its ads
+
+  BEGIN;
+  DELETE from campaigns where id = 46 AND company_id = 5;
+  DELETE from ads where campaign_id = 46 AND company_id = 5;
+  COMMIT;
+
+When people scale applications with NoSQL databases they often miss the lack of transactions and joins. We already saw a join query when discussing distribution columns, but here's another to combine information from campaigns and ads.
+
+::
+
+  -- Total campaign budget vs expense this month
+
+  SELECT camp.monthly_budget,
+         CASE cost_model
+         WHEN 'cost_per_click' THEN
+           clicks_count
+         ELSE
+           impressions_count
+         END AS billable_events
+  FROM campaigns camp
+  JOIN ads a ON (
+        a.campaign_id = camp.id
+    AND a.company_id = camp.company_id)
+  WHERE camp.company_id = 5;
+
+Sharing Data Between Tenants
+----------------------------
+
+Up until now all tables have been distributed by :code:`company_id`, but sometimes there is data that can be shared by all tenants, and doesn't "belong" to any tenant in particular. For instance all companies using this example ad platform might want to get geographical information for their audience based on IP addresses. In a single machine database this could be accomplished by a lookup table for geo-ip, like the following. (A real table would probably use PostGIS but bear with the simplified example.)
+
+::
+
+  CREATE TABLE geo_ips (
+    addrs cidr NOT NULL PRIMARY KEY,
+    latlon point NOT NULL
+      CHECK (-90  <= latlon[0] AND latlon[0] <= 90 AND
+             -180 <= latlon[1] AND latlon[1] <= 180)
+  );
+  CREATE INDEX ON geo_ips USING gist (addrs inet_ops);
+
+Joining clicks with this table can produce, for example, the locations of everyone who clicked on ad 456.
+
+::
+
+  SELECT latlon
+    FROM geo_ips, clicks c
+   WHERE addrs >> c.user_ip
+     AND c.clicked_at > current_date - INTERVAL '1 day'
+     AND c.company_id = 5
+     AND c.ad_id = 456;
+
+In Citus we need to find a way to co-locate the :code:`geo_ips` table with clicks for every company. One way would be to add a :code:`company_id` column to :code:`geo_ips` and duplicate the data in the table for every company. This approach is not optimal because it introduces the burden of keeping the data synchronized between the companies if and when it changes. A more convenient way is by designating :code:`geo_ips` as a *reference table.*
+
+Reference tables in Citus have exactly one shard, and it is replicated across all worker nodes. This co-locates the information for all tenants' queries. It does require reserving space for a copy of the data on all nodes, but automatically stays in sync during reference table updates. To make a reference table, call :ref:`create_reference_table <create_reference_table>` for a table on the coordinator node:
```

Comment:
What do you think about replacing "It does require reserving space for a copy of the data on all nodes, but automatically stays in sync during reference table updates."

with something like below (with better wording)

"Reference tables in Citus automatically stay in sync with modifications as Citus makes sure to forward them to all copies of the shard consistently."